rs/boundary_node/ic_boundary/src/routes.rs

```diff
@@ -166,45 +148,110 @@ impl ErrorCause {
-            Self::NoHealthyNodes => write!(f, "no_healthy_nodes"),
-            Self::ReplicaErrorDNS(_) => write!(f, "replica_error_dns"),
-            Self::ReplicaErrorConnect => write!(f, "replica_error_connect"),
-            Self::ReplicaTimeout => write!(f, "replica_timeout"),
-            Self::ReplicaTLSErrorOther(_) => write!(f, "replica_tls_error"),
-            Self::ReplicaTLSErrorCert(_) => write!(f, "replica_tls_error_cert"),
-            Self::ReplicaErrorOther(_) => write!(f, "replica_error_other"),
-            Self::RateLimited(x) => write!(f, "rate_limited_{x}"),
+            Self::Other(_) => ErrorClientFacing::Other,
+            Self::BodyTimedOut => ErrorClientFacing::BodyTimedOut,
+            Self::UnableToReadBody(x) => ErrorClientFacing::UnableToReadBody(x.clone()),
+            Self::PayloadTooLarge(x) => ErrorClientFacing::PayloadTooLarge(*x),
+            Self::UnableToParseCBOR(x) => ErrorClientFacing::UnableToParseCBOR(x.clone()),
+            Self::UnableToParseHTTPArg(x) => ErrorClientFacing::UnableToParseHTTPArg(x.clone()),
+            Self::LoadShed => ErrorClientFacing::LoadShed,
+            Self::MalformedRequest(x) => ErrorClientFacing::MalformedRequest(x.clone()),
+            Self::MalformedResponse(_) => ErrorClientFacing::ReplicaError,
+            Self::NoRoutingTable => ErrorClientFacing::ServiceUnavailable,
+            Self::SubnetNotFound => ErrorClientFacing::SubnetNotFound,
+            Self::CanisterNotFound => ErrorClientFacing::CanisterNotFound,
+            Self::NoHealthyNodes => ErrorClientFacing::NoHealthyNodes,
+            Self::ReplicaErrorDNS(_) => ErrorClientFacing::ReplicaError,
+            Self::ReplicaErrorConnect => ErrorClientFacing::ReplicaError,
+            Self::ReplicaTimeout => ErrorClientFacing::ReplicaError,
+            Self::ReplicaTLSErrorOther(_) => ErrorClientFacing::ReplicaError,
+            Self::ReplicaTLSErrorCert(_) => ErrorClientFacing::ReplicaError,
+            Self::ReplicaErrorOther(_) => ErrorClientFacing::ReplicaError,
+            Self::RateLimited(_) => ErrorClientFacing::RateLimited,
         }
     }
 }
 
 // Creates the response from ErrorCause and injects itself into extensions to be visible by middleware
 impl IntoResponse for ErrorCause {
     fn into_response(self) -> Response {
-        let mut body = self.to_string();
+        let client_facing_error = self.to_client_facing_error();
+        client_facing_error.into_response()
+    }
+}
+
+#[derive(Clone, Debug, Display, IntoStaticStr)]
+#[strum(serialize_all = "snake_case")]
+pub enum ErrorClientFacing {
+    BodyTimedOut,
+    CanisterNotFound,
+    LoadShed,
+    MalformedRequest(String),
+    NoHealthyNodes,
+    #[strum(serialize = "internal_server_error")]
+    Other,
+    PayloadTooLarge(usize),
+    RateLimited,
+    ReplicaError,
+    ServiceUnavailable,
+    SubnetNotFound,
+    UnableToParseCBOR(String),
+    UnableToParseHTTPArg(String),
+    UnableToReadBody(String),
```

Comment:
It seems that `UnableToReadBody` is only happening when we can't read the client's request. This can happen only when the connection is broken or timed out, so we can't really provide this error to the client. 

So probably we shouldn't designate it as client-facing